### PR TITLE
Scripts: VTMD table [closes #122]

### LINKED ISSUE #122: VTMD InnoDB system table
InnoDB system table analogous to VTQ. Information schema for table for tracking data and debug. Table structure is described [here](https://github.com/tempesta-tech/mariadb_10.2/wiki/DDL-Survival#vtd-structure).

### PR DESCRIPTION
**Pull request by kevgs**
Just one primary index because I don't know access pattern to this table.

### CODE REVIEW

**Review by midenok — CHANGES_REQUESTED** [review]
Full test failures. See build 68 on Sunny.

**Comment by midenok** [comment]
Will upgrade work from older version?

**Comment by midenok** [comment]
Rename `innodb_vtd` to `vtmd`.

File: scripts/mysql_system_tables_fix.sql [hunk 1]
```diff
@@ -699,6 +699,20 @@ PREPARE stmt FROM @str;
...
 
+SET @create_vtmd="CREATE TABLE IF NOT EXISTS vtmd (
+	trx_id_start			BIGINT UNSIGNED NOT NULL,
+	trx_id_end			BIGINT UNSIGNED NOT NULL,
+	old_name			VARCHAR(64),
+	name				VARCHAR(64) NOT NULL,
+	frm_image			BLOB,
+	col_renames			BLOB,
+	PRIMARY KEY(trx_id_start, trx_id_end, name)
+) ENGINE=INNODB DEFAULT CHARSET=utf8 COLLATE=utf8_bin STATS_PERSISTENT=0";
```
Comment (midenok) [inline scripts/mysql_system_tables_fix.sql]: Do we have to duplicate creation command?

File: mysql-test/r/1st.result [hunk 2]
```diff
@@ -37,3 +37,4 @@ time_zone_name
 time_zone_transition
```
Comment (midenok) [inline mysql-test/r/1st.result:37]: Abbreviations are written in all caps in commit messages.

File: scripts/mysql_system_tables.sql [hunk 3]
```diff
@@ -129,6 +129,16 @@ SET @create_innodb_index_stats="CREATE TABLE IF NOT EXISTS innodb_index_stats (
...
 ) ENGINE=INNODB DEFAULT CHARSET=utf8 COLLATE=utf8_bin STATS_PERSISTENT=0";
 
+SET @create_vtmd="CREATE TABLE IF NOT EXISTS vtmd (
+	trx_id_start			BIGINT UNSIGNED NOT NULL,
+	trx_id_end			BIGINT UNSIGNED NOT NULL,
+	old_name			VARCHAR(64),
+	name				VARCHAR(64) NOT NULL,
+	frm_image			BLOB,
+	col_renames			BLOB,
+	PRIMARY KEY(trx_id_start, trx_id_end, name)
```
Comment (midenok) [inline scripts/mysql_system_tables.sql]: Don't need `trx_id_end` in PK. Read about [Multiple column index vs multiple indexes](https://www.percona.com/blog/2014/01/03/multiple-column-index-vs-multiple-indexes-with-mysql-56/) to know better on the topic.

**Review by midenok — CHANGES_REQUESTED** [review]
Remove `trx_id_end` from clustered index and fix minor notes. Run full test on Sunny.

File: scripts/mysql_system_tables.sql (hunk 3 above)
Comment (midenok) [reply scripts/mysql_system_tables.sql]: Please, comment each column.

**Comment by midenok** [comment]
Rename `trx_id_start` to `start`, `trx_id_end` to `end`. Make VTMD versioned table by period (start, end).

**Comment by kevgs** [comment]
>  Will upgrade work from older version?

Yes, it will.

> Rename innodb_vtd to vtmd.

Done.

> Remove trx_id_end from clustered index and fix minor notes. Run full test on Sunny.

> Do we have to duplicate creation command?

Actually not, because `mysql_system_tables.sql` and `mysql_system_tables_fix.sql` got concatenated in a single `mysql_fix_privilege_tables.sql` which is used by `mysql_upgrade`

> Abbreviations are written in all caps in commit messages.

Done.

> Don't need trx_id_end in PK.

Done.

> Please, comment each column.

Done.

> Rename trx_id_start to start, trx_id_end to end. Make VTMD versioned table by period (start, end).

Done.


Currently waiting for sunny.

**Comment by midenok** [comment]
Comments are marked by `COMMENT` clause, not by #-sign.

**Comment by kevgs** [comment]
Columns with `GENERATED ALWAYS AS ROW START/END` do not support `COMMENT 'asdf'`.

**Comment by midenok** [comment]
VTMD is not created on start. Did you test upgrade?

**Comment by midenok** [comment]
> Columns with GENERATED ALWAYS AS ROW START/END do not support COMMENT 'asdf'.

Why?

**Comment by kevgs** [comment]
For my `VTMD` is created by both `mysql_install_db` and `mysql_upgrade`. How did you tested it?

> > Columns with GENERATED ALWAYS AS ROW START/END do not support COMMENT 'asdf'.

> Why?

It's not implemented in sql/sql_yacc.yy. Looks like a miss feature.

**Comment by midenok** [comment]
> It's not implemented in sql/sql_yacc.yy. Looks like a miss feature.

It's just a bug. Fix it.

File: mysql-test/suite/versioning/r/ddl.result [hunk 4]
```diff
@@ -161,6 +161,19 @@ drop procedure concat_exec2;
 drop procedure concat_exec3;
 drop function get_historical_table_name;
 drop procedure drop_last_historical;
+select * from mysql.vtmd;
+start	end	old_name	name	frm_image	col_renames
+show create table mysql.vtmd;
+Table	Create Table
+vtmd	CREATE TABLE `vtmd` (
+  `start` bigint(20) unsigned GENERATED ALWAYS AS ROW START COMMENT 'transaction which created table',
```
Comment (midenok) [inline mysql-test/suite/versioning/r/ddl.result]: TRX_ID of table lifetime start

File: mysql-test/suite/versioning/r/ddl.result [hunk 5]
```diff
@@ -161,6 +161,19 @@ drop procedure concat_exec2;
...
+  `end` bigint(20) unsigned GENERATED ALWAYS AS ROW END COMMENT 'transaction which altered table',
```
Comment (midenok) [inline mysql-test/suite/versioning/r/ddl.result]: TRX_ID of table lifetime end

File: mysql-test/suite/versioning/r/ddl.result [hunk 6]
```diff
@@ -161,6 +161,19 @@ drop procedure concat_exec2;
...
+  `old_name` varchar(64) COLLATE utf8_bin DEFAULT NULL COMMENT 'old table name if it was renamed',
+  `name` varchar(64) COLLATE utf8_bin NOT NULL COMMENT 'current table name',
```
Comment (midenok) [inline mysql-test/suite/versioning/r/ddl.result]: Table name during period [start, end)

File: mysql-test/suite/versioning/r/ddl.result [hunk 7]
```diff
@@ -161,6 +161,19 @@ drop procedure concat_exec2;
...
+  `frm_image` blob DEFAULT NULL COMMENT '.frm file contents',
```
Comment (midenok) [inline mysql-test/suite/versioning/r/ddl.result]: Table structure during period [start, end)
Make NOT NULL

File: mysql-test/suite/versioning/r/ddl.result [hunk 8]
```diff
@@ -161,6 +161,19 @@ drop procedure concat_exec2;
...
+  `col_renames` blob DEFAULT NULL COMMENT 'column rename mappings',
```
Comment (midenok) [inline mysql-test/suite/versioning/r/ddl.result]: Column name mapping from previous lifetime

File: mysql-test/suite/versioning/r/ddl.result [hunk 9]
```diff
@@ -161,6 +161,19 @@ drop procedure concat_exec2;
...
 drop function get_historical_table_name;
 drop procedure drop_last_historical;
+select * from mysql.vtmd;
+start	end	old_name	name	frm_image	col_renames
+show create table mysql.vtmd;
+Table	Create Table
+vtmd	CREATE TABLE `vtmd` (
+  `start` bigint(20) unsigned GENERATED ALWAYS AS ROW START COMMENT 'transaction which created table',
+  `end` bigint(20) unsigned GENERATED ALWAYS AS ROW END COMMENT 'transaction which altered table',
+  `old_name` varchar(64) COLLATE utf8_bin DEFAULT NULL COMMENT 'old table name if it was renamed',
```
Comment (midenok) [inline mysql-test/suite/versioning/r/ddl.result]: Remove this column.

File: mysql-test/suite/versioning/r/ddl.result [hunk 10]
```diff
@@ -161,6 +161,19 @@ drop procedure concat_exec2;
 drop procedure concat_exec3;
 drop function get_historical_table_name;
 drop procedure drop_last_historical;
+select * from mysql.vtmd;
+start	end	old_name	name	frm_image	col_renames
+show create table mysql.vtmd;
+Table	Create Table
+vtmd	CREATE TABLE `vtmd` (
```
Comment (midenok) [inline mysql-test/suite/versioning/r/ddl.result]: Rename to `vtmd_template`.

**Review by midenok — APPROVED** [review]
Fix comments, rename table, remove `old_name`, make `frm_image` NOT NULL and merge. Don't forget to rerun main suite.

**Comment by kevgs** [comment]
`innodb.log_data_file_size` do not pass without some `PRIMARY KEY` on `vtmd_template`. Adding key on (start, end).

**Comment by midenok** [comment]
Why do you make such strange PK? Make it on (end).

**Pull request merged**